src/test/java/gov/healthit/chpl/aqa/stepDefinitions/Hooks.java

```diff
@@ -1,147 +1,166 @@
+				// If temp directory creation failed, create new directory in target folder
+				// user.dir - User working directory, make new directories in user's working directory
+				File file = new File("target", "download-files");
+				file.mkdirs();
+				tempDirectory = System.getProperty("user.dir") + File.separator + "target" + File.separator + "download-files";
+			}
+			downloadPath = tempDirectory;
+		}
+		dir = new File(downloadPath);
+		if (!dir.exists()) {
+			dir.mkdirs();
+		}
+
+		if (config.getProperty("browser").equals("chrome")) {
+			HashMap<String, Object> chromePrefs = new HashMap<String, Object>();
+			chromePrefs.put("profile.default_content_settings.popups", 0);
+			chromePrefs.put("download.default_directory", downloadPath);
+			/**
+			 * Save Chrome Options
+			 */
+			ChromeOptions options = new ChromeOptions();
+			options.setExperimentalOption("prefs", chromePrefs);
+			chromePrefs.put("safebrowsing.enabled", "true");
+			options.addArguments("--safebrowsing-disable-download-protection");
+			options.addArguments("disable-popup-blocking");
+			options.setCapability(CapabilityType.ACCEPT_SSL_CERTS, true);
+			screenshotPath = System.getProperty("user.dir") + File.separator + "test-output";
+			driver = new EventFiringWebDriver(new ChromeDriver(options));
+		}
+
+		else if(config.getProperty("browser").equals("firefox")) {
+			driver = new EventFiringWebDriver(new FirefoxDriver());
+		}
+
+		else if(config.getProperty("browser").equals("edge")) {
+			driver =new EventFiringWebDriver( new EdgeDriver());
+		}
+
+		driver.manage().timeouts().implicitlyWait(DELAY, TimeUnit.SECONDS);
+		WebDriverEventListener errorListener = new AbstractWebDriverEventListener() {
+			@Override
+			public void onException(final Throwable throwable, final WebDriver activeDriver) {
+				try {
+					takeScreenshot();
+				} catch (Exception e) {
+					e.printStackTrace();
+				}
+			}
+		};
+		driver.register(errorListener);
+	}
+
+	/**
+	 * Close browser windows and terminate WebDriver session.
+	 */
+	@After("~@RegressionAPI")
+	public void afterMethod() {
+		driver.quit();
+	}
+
+	public static EventFiringWebDriver getDriver() {
+		return driver;
+	}
+
+	public static File getDownloadDirectory() {
+		return dir;
+	}
+
+	/**
+	 * Take a screenshot.
+	 * @param hash a string that will be inserted into the filename to avoid overwriting images.
+	 * All "non-word characters" will be replaced with "_"
+	 * @throws Exception if there is an exception
+	 */
+	public static void takeScreenshot(final String hash) throws Exception {
+		File scrFile = ((TakesScreenshot) getDriver()).getScreenshotAs(OutputType.FILE);
+		File outFile = new File(screenshotPath + File.separator + "failed-test-" + hash.replaceAll("\\W+", "_") + ".png");
+		FileHandler.copy(scrFile, outFile);
+	}
+
+	/**
+	 * Take a screenshot. Generates hash based on current time.
+	 * @throws Exception if screenshot cannot be created
+	 */
+	public static void takeScreenshot() throws Exception {
```

Comment:
I don't know why all of these lines are showing as changed. Is it because of the change from spaces to tabs? That's making it hard to tell what actually changed

Reply:
nothing changed here. its checkstyle